services-hll/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/RequestItem.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.model;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Consumer;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.NotThreadSafe;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.enhanced.dynamodb.AsyncTable;
+import software.amazon.awssdk.enhanced.dynamodb.Table;
+import software.amazon.awssdk.enhanced.dynamodb.converter.ItemAttributeValueConverter;
+import software.amazon.awssdk.enhanced.dynamodb.internal.model.DefaultRequestItem;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * An item that can be sent to DynamoDB. An item is a single, unique entry in a DynamoDB table.
+ *
+ * A {@code RequestItem} is a {@code Map<String, Object>} that can be converted into a {@code Map<String, ItemAttributeValue>}
+ * using the configured {@link #converters()}.
+ *
+ * @see Table
+ * @see AsyncTable
+ */
+@SdkPublicApi
+@ThreadSafe
+@Immutable
+public interface RequestItem extends ConverterAware,
+                                     AttributeAware<Object>,
+                                     ToCopyableBuilder<RequestItem.Builder, RequestItem> {
+    /**
+     * Create a builder that can be used for configuring and creating a {@link RequestItem}.
+     */
+    static Builder builder() {
+        return DefaultRequestItem.builder();
+    }
+
+    /**
+     * Convert this request item into a {@link GeneratedRequestItem} using the converters attached to this item, only.
```

Comment:
why only use the converters attached to this item?

Reply:
RequestItems are not associated with any other context when they're created.

How would this example work with any converter OTHER than foo?

```Java
RequestItem.builder()
           .putAttribute("id", "123")
           .addConverter(foo)
           .build()
           .toGeneratedRequestItem()
```

Reply:
Makes sense